### ISSUE TITLE
Corner Case - Banner size limit

### ISSUE THREAD

**Issue by mhrimaz** [se2p/programming-styles-ss-2020-program-description#1]
Imagine this state of the game, where we don't have a lot of space to show the banners and messages, for example, the invalid input message is too long to fit it into this area.

```
  ╔════════╤═══╤═══════╗
  ║  ┌──┐1 │   │   2   ║
  ║  │  │  │ 3 │       ║
  ║  │4 │  │   │       ║
  ║  └──┘6 │ 5 │       ║
  ╚════════╧═══╧═══════╝
```
Then what should we do in this state? Maybe you would suggest increasing the height of the banner. So we can convert this one:
```
┌────────────────────────────┐
│ The fishing was good; it's │
│ the catching that was bad. │
└────────────────────────────┘
```
to this one:
```
  ╔════════╤═══╤═══════╗
 ┌──────────────────────┐
 │ The fishing was good;│
 │   it's the catching  │
 │     that was bad.    │
 └──────────────────────┘
  ╚════════╧═══╧═══════╝
```
But this is not going to work in all cases, especially for the long invalid input message. As an alternative we can fix the length of our board by drawing the river tiles behind the boat, for example, something like this one:
```
  ╔════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
  ║  ┌──┐1 │   │   │   │   │   │ 2 │   │   │   │   │   │       ║ 
  ║  │  │  │   │   │   │   │   │ 3 │   │   │   │   │   │       ║
  ║  │  │  │   │   │   │   │   │ 4 │   │   │   │   │   │       ║
  ║  └──┘6 │   │   │   │   │   │ 5 │   │   │   │   │   │       ║ 
  ╚════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
  
  input: [1,1,1,2,2,2,2]
  ╔═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
  ║   │   │   │  ┌──┐1 │   │   │   │   │   │ 2 │   │   │       ║ 
  ║   │   │   │  │  │  │   │   │ 3 │   │   │   │   │   │       ║
  ║   │   │   │  │  │  │   │   │ 4 │   │   │   │   │   │       ║
  ║   │   │   │  └──┘6 │   │   │ 5 │   │   │   │   │   │       ║ 
  ╚═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
```

Don't you think that printing a line before or after the board is a more viable solution, and it doesn't hurt the point of assignment? I think mimicking the described output, especially printing the banner, is an overkill.

**Comment by alessiogambi** [comment]
In case the board is too short, the message banner will be simply longer than it. The important thing is that the banner **must be** centered over the board as shown below.  The height of the banner must always be the same and is the number of little tiles that changes with the game.

Note that the status of the board is the one **after the input is processed** so your example is inconsistent (the little tile should not be there).
```
      ╔════════╤═══════╗
┌────────────────────────────┐
│ The fishing was good; it's │
│ the catching that was bad. │
└────────────────────────────┘
      ╚════════╧═══════╝
```

@mhrimaz Please can create a pull request that includes this corner case in the program description?

**Comment by mhrimaz** [comment]
Thanks for your clarification.

> Note that the status of the board is the one after the input is processed so your example is inconsistent (the little tile should not be there).

My bad.

> @mhrimaz Please can create a pull request that includes this corner case in the program description?

@alessiogambi Sure, I am working on it.

**Comment by mhrimaz** [comment]
@alessiogambi One further question:

> In case the board is too short, the message banner will be simply longer than it. The important thing is that the banner must be centered over the board as shown below. The height of the banner must always be the same and is the number of little tiles that changes with the game.

I noticed that your board drawing is inconsistent during the program description. Can you clarify which one of the following is the correct one?
Imagine this board (the printed number on the left column is just for clarification):
```
1  ╔════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
2  ║  ┌──┐1 │   │   │   │   │ 2 │   │   │   │   │   │       ║ 
3  ║  │ 3│  │   │   │   │   │   │   │   │   │   │   │       ║
4  ║  │4 │  │   │   │   │   │   │   │   │   │   │   │       ║
5  ║  └──┘6 │   │   │   │   │ 5 │   │   │   │   │   │       ║ 
6  ╚════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
```
We want to print the invalid input message on this board. The question is, which one is correct?
A (this is what you have drawn in your description):
```
1  ╔════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
2  ║  ┌──┐1 │   │   │   │   │ 2 │   │   │   │   │   │       ║ 
3 ┌──────────────────────────────────────────────────────────┐
4 │             The provided input is not valid!             │
5 │                 Offending input: 400...                  │
6 └──────────────────────────────────────────────────────────┘
7  ║  └──┘6 │   │   │   │   │ 5 │   │   │   │   │   │       ║ 
8  ╚════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝

```

B:
```
1  ╔════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
2 ┌──────────────────────────────────────────────────────────┐
3 │             The provided input is not valid!             │
4 │                 Offending input: 400...                  │
5 └──────────────────────────────────────────────────────────┘
6  ╚════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝

```

**Comment by alessiogambi** [comment]
**B** is the correct one, let me fix that.

**Comment by alessiogambi** [comment]
Done via #7

**Comment by alessiogambi** [comment]
This is closed by #8
